cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -482,9 +484,17 @@ private static String getCdapCommonJarName() {
         LOG.warn("Failed to locate cdap-common.jar. It will not be added to the spark extra classpath in the" +
                    " beginning.");
       }
+
+      List<String> extraJarPaths = new ArrayList<>();
+      // Add extra jars set in cConf to jarFiles
+      for (String jar : LocalizationUtils.getExtraJarNames(cConf)) {
```

Comment:
Same as above. Shouldn't expect two different calls to a util method gives you matching jar file and jar name. Should always rely on the result of `addLocalizedExtraJars`.